### DISCUSSION TITLE
Maximum town bridge length

### DISCUSSION THREAD

**Discussion by 2TallTyler** [OpenTTD/OpenTTD#8388] [General]
![bridgeport](https://user-images.githubusercontent.com/55058389/102298372-ffe99900-3f1e-11eb-8e94-650d570bc7e2.png)

While poking around [GrowTownWithBridge()](https://github.com/OpenTTD/OpenTTD/blob/master/src/town_cmd.cpp#L1100), I discovered that the lengths (not including bridge ends) are capped by magic numbers at 3 tiles for a ramped bridge over tracks or rivers, and 10 tiles for a flat bridge over ocean.

EDIT 12/17: 
Fixed these numbers from my original post which said 4 and 11. The code counts the bridge length from start tile, counting the end tile but not the start tile.

With this knowledge, I actually propose increasing the ramped bridge limit to 5 tiles to allow crossing 4 tracks.
/END EDIT

I think 10 tiles is unnecessarily long and leads only to silliness. It's been this way since at least 2007, which is as far as I can go on GitHub, so I'm not sure how this number was chosen.

I generated a map with lots of bridges (high number of towns and medium sea level) and did a quick survey of bridge lengths (again, not counting bridge ends). In general, those with a length <= 4  nearly always look appropriate, bridges length == 5 are about evenly split between silly and appropriate, and bridges > 5 tiles are mostly silly.

I can't think of any negative effects to lowering this limit to either 4 or 5, but that's why I'm starting this discussion before I make a PR changing this to a number I've unilaterally chosen. Thoughts?

**Comment by andythenorth** [comment]
+1

**Comment by auge8472** [comment]
-1 Your proposed limits might match your play style but the style of other players will differ.

**Reply by 2TallTyler** [reply]
True. I would note that this only affects town generation and growth, and not player-built bridges. How might a player use town-built bridges in their play style? (genuinely curious, not challenging you)

**Comment by auge8472** [comment]
> I would note that this only affects town generation and growth, and not player-built bridges.

Got me. After rereading your proposal (especially the heading that explicitely states the circumstance of being about town built bridges :-)) it is o.k. for me to think about lowering the limit of the sea-bridges length. In [insert the bad R-word here] [1] we'll find several examples of long bridges overspanning wide bayes. A maximum length of five tiles results in a six tiles long bridge (with the (flat) ramps), overspanning a width of four water tiles.

Might be a compromise between preventing (subjective felt) uglyness and initial town expansion possibilities. At least the maximum length should be an odd (uneven?) number so that the longest bridges have an even length (with the ramps). That alone would lead to a nicer look for many cases.

[1]: reality
